smoderp2d/providers/base/data_preparation.py

```diff
@@ -167,28 +167,27 @@ def _get_rr_rc(r, c, mat_boundary):
         rr = []
         rc = []
 
-        in_domain = False
         in_boundary = False
+        rr_insert = False
 
         for i in nr:
             one_col = []
-            one_col_boundary = []
             for j in nc:
 
                 if mat_boundary[i][j] == -99 and in_boundary is False:
                     in_boundary = True
 
                 if mat_boundary[i][j] == -99 and in_boundary is True:
-                    one_col_boundary.append(j)
-
-                if (mat_boundary[i][j] == 0.0) and in_domain is False:
-                    rr.append(i)
-                    in_domain = True
+                    one_col.append(j)
+                    rr_insert = True
```

Comment:
I actually think that instead of:

```
                if mat_boundary[i][j] == -99 and in_boundary is False:
                    in_boundary = True

                if mat_boundary[i][j] == -99 and in_boundary is True:
                    one_col.append(j)
```

We can use the following:

```
                if mat_boundary[i][j] == -99:
                    one_col.append(j)
```

And then get rid of `in_boundary`. (maybe just put it there as a comment) 

@landam: What do you think? I remember that we have discussed that and have decided not to do so, but I don't see any reasonably reasonable reasoning behind that now. Do you? 

PS: The next `if` could then be changed to `elif`.